src/acl/RegexData.cc

```diff
@@ -113,45 +104,28 @@ removeUnnecessaryWildcards(char * t)
     return t;
 }
 
-static bool
-compileRE(std::list<RegexPattern> &curlist, const char * RE, int flags)
+static void
+compileRE(std::list<RegexPattern> &curlist, const SBuf &RE, int flags)
 {
-    if (RE == NULL || *RE == '\0')
-        return curlist.empty(); // XXX: old code did this. It looks wrong.
```

Comment:
I removed "XXX: old code did this. It looks wrong" code/comments during this refactoring. The comment (added in commit e2b7452) can be considered correct: Out of the 3-4 situations where aggregated REs can be empty, the "old" odd-looking code handled only cases 1 and 4.

1. "-i/+i" at the beginning of the ACL data: No accumulated REs yet.
2. "-i/+i" after a buffering/joining overflow emptied accumulated REs.
3. `regex1 +i -i regex2` and similar sequences: Accumulated REs (regex1) are emptied upon seeing "+i". Squid then tries to compile still-empty accumulated REs upon seeing the "-i".
4. An empty RE token. This may not be possible with recent RE parsers.

PR code does not call compileREs() with an empty list of REs, handling situations 1-3. I let the regex library handle the empty (zero-length) RE token (case 4) instead of treating it as a special case. Again, I doubt that use case is actually real.

Reply:
Case 4 can occur when the parser state implies that a 0-length string is the token for pattern. Real-world situations usually involve multi-file configs with the values and ordering spread out, so not as obviously silly as this test case:
```
configuration_includes_quoted_values on
acl regex dstdomain_regex ""
refresh_pattern "" 0 0 0
```

A unit test to check the library behaviour seems warranted if you don't want to add logic to detect and throw a config error on such madness.

Reply:
I do not want to work on handling empty regexes _specially_ at least until there is some evidence that it is possible to create an empty regex. I knew that empty configuration tokens are possible in certain contexts, but the example you sketched does not work because `configuration_includes_quoted_values on` cases are officially excluded from handling regexes:

```C++
if (RecognizeQuotedValues)
    throw TextException("Cannot read regex expression while configuration_includes_quoted_values is enabled", Here());
```